akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -216,10 +216,11 @@ private[http] object HttpServerBluePrint {
       case start: RequestStart ⇒
         try {
           val effectiveUri = HttpRequest.effectiveUri(start.uri, start.headers, isSecureConnection, defaultHostHeader)
+          HttpRequest.verifyUri(effectiveUri)
```

Comment:
If we verify it here we might want to remove the other check in `HttpRequest` for performance reasons. Or, we don't verify here and catch and handle the exception at the place where the request is created. I guess handling it at the place where the HttpRequest is created would be simpler to achieve.